sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/firestore/FirestoreBatchRequest.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.beam.sdk.io.gcp.firestore;
+
+import com.google.cloud.firestore.DocumentReference;
+import com.google.cloud.firestore.Firestore;
+import com.google.cloud.firestore.WriteBatch;
+
+import java.util.List;
+
+
+public class FirestoreBatchRequest<T> {
+    private Firestore firestoreClient;
+
+    public FirestoreBatchRequest(Firestore firestoreClient) {
+        this.firestoreClient = firestoreClient;
+    }
+
+    public WriteBatch batchWithKey(List<T> input, String collection, String key) {
+        WriteBatch batch = firestoreClient.batch();
```

Comment:
Please note that `WriteBatch` is atomic transaction.
Large `WriteBatch` could leads to contention and high error rates.

We are working on launching a batchWrite API for non-atomic data ingestion use cases. Before it gets launched, the next best option is writing each document separately.

(BTW, we have a max 500 write per commit limit.)

Reply:
Please make sure this is addressed.